packages/runtime/container-runtime/src/runCounter.ts

```diff
@@ -0,0 +1,25 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export class RunCounter {
```

Comment:
@markfields thinking i'll bring this to main. i like it better for orderSequentially and re-entrancy, as it reduces mutable state in the container runtime, and i think makes all the code easier to follow.

Reply:
https://github.com/microsoft/FluidFramework/pull/24134